nix/module.nix

```diff
@@ -0,0 +1,66 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.services.tuxedo-rs;
+in {
+  options = {
+    services.tuxedo-rs = {
+      enable = mkEnableOption ''
+        Rust utilities for interacting with hardware from TUXEDO Computers.
+      '';
+
+      tailor_gui.enable = mkEnableOption ''
+        Alternative to Tuxedo Control Center, written in Rust.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    hardware.tuxedo-keyboard.enable = true;
+
+    systemd = {
+      services.tailord = {
+        enable = lib.mkDefault true;
+        description = "Tux Tailor hardware control service";
+        after = ["systemd-logind.service"];
+        wantedBy = ["multi-user.target"];
+
+        serviceConfig = {
+          Type = "dbus";
+          BusName = "com.tux.Tailor";
+          ExecStart = "${pkgs.tailord}/bin/tailord";
+          Environment = "RUST_BACKTRACE=1";
+          Restart = "on-failure";
+        };
+      };
+    };
+
+    services.dbus.packages = [pkgs.tailord];
+
+    # NOTE: By setting mode, the files are copied and not symlinked
+    environment = {
+      etc = {
+        "tailord/keyboard/default.json" = {
+          source = ../tailord/default_configs/keyboard/default.json;
+          mode = "644";
+        };
+        "tailord/fan/default.json" = {
+          source = ../tailord/default_configs/fan/default.json;
+          mode = "644";
+        };
+        "tailord/profiles/default.json" = {
+          source = ../tailord/default_configs/profiles/default.json;
+          mode = "644";
+        };
+      };
```

Comment:
I'm not sure what happens if these default files are modified in tailor_gui and then this module is rebuilt.
We probably wouldn't want them to be overwritten, but we also wouldn't want NixOS to output an error message if they have been modified.

In any case, I think it might be better if tailord creates or copies (from a read-only location) the default settings at startup if they don't exist.

- [ ] TODO: Test this.

Reply:
Seems like they are overwritten. The default profile's performance profile is unset after a rebuild.